lib/carrier/messaging/tracker.ex

```diff
@@ -0,0 +1,179 @@
+defmodule Carrier.Messaging.Tracker do
+
+  defstruct [reply_endpoints: %{}, subscriptions: %{}, monitors: %{}, unused_topics: []]
+
+  @spec add_reply_endpoint(Tracker.t, pid()) :: {Tracker.t, String.t}
+  def add_reply_endpoint(%__MODULE__{reply_endpoints: reps}=tracker, subscriber) do
+    case Map.get(reps, subscriber) do
+      nil ->
+        topic = make_reply_endpoint_topic()
+        reps = reps
+        |> Map.put(subscriber, topic)
+        |> Map.put(topic, subscriber)
+        {maybe_monitor_subscriber(%{tracker | reply_endpoints: reps}, subscriber), topic}
+      topic ->
+        {tracker, topic}
```

Comment:
Is it a valid scenario to call this function more than once per subscriber, or is this a defensive, just-in-case thing?